packages/blockchain-wallet-v4/src/redux/webSocket/bch/sagas.js

```diff
@@ -0,0 +1,47 @@
+import { call, put, select } from 'redux-saga/effects'
+import { compose } from 'ramda'
+import * as A from '../../actions'
+import * as walletSelectors from '../../wallet/selectors'
+import { Socket } from '../../../network/index'
+import * as bchActions from '../../data/bch/actions'
+
+// TO REVIEW
+export default ({ api, bchSocket }) => {
+  const send = bchSocket.send.bind(bchSocket)
+
+  const onOpen = function * () {
+    const subscribeInfo = yield select(walletSelectors.getInitialSocketContext)
+    yield call(compose(send, Socket.onOpenMessage), subscribeInfo)
+  }
+
+  const onMessage = function * (action) {
+    const message = action.payload
+
+    switch (message.op) {
+      case 'utx':
+        yield put(A.webSocket.bitcoin.paymentReceived('You\'ve just received a Bitcoin Cash payment.'))
```

Comment:
A bit weird we're calling paymentReceived on bitcoin here and in eth saga, but okay since it ends up just dispatching the showAlert action.